Towers/main.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const assert = require('assert');
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+let stacks = {
+  a: [4, 3, 2, 1],
+  b: [],
+  c: []
+};
+
+const printStacks = () => {
+  console.log("a: " + stacks.a);
+  console.log("b: " + stacks.b);
+  console.log("c: " + stacks.c);
+}
+/*this function takes in the startStack name and endStack name, and moves the top piece 
+from the corresponding starting stack to the corresponding ending stack*/
+const movePiece = (startStack, endStack) => {
+  // Your code here
+  let moveStack = stacks[startStack]
+  let twoStack = stacks[endStack]
+  let piece = moveStack.pop(); // piece equals the last index of the choosen array
+  twoStack.push(piece); // twoStack is pushing the piece to the array that was choosen  
+  //twoStack.push(moveStack.pop())
+  console.log(piece)
+
+}
+
+/*This function takes in the start stack name, and end stack name,
+ and returns true only if the move is a legal and valid move.
+ Otherwise it returns false*/
+const isLegal = (startStack, endStack) => {
+  // Your code here
+  let moveStack = stacks[startStack] // stacks a
+  let twoStack = stacks[endStack] //stack b
+  let moveStackPiece = moveStack[moveStack.length - 1] // stack at "a" will be the last index
+  let twoStackPiece = twoStack[twoStack.length -1]  // stack at "b" will be the last index
+  console.log(twoStackPiece,"twostackpiece")
+  //When moving a piece to a postion that is blank the computer does know no less the postion is undefined
+  if(moveStackPiece < twoStackPiece || twoStackPiece == undefined){
+    return true
+  }
+  else{
+    return false
+  }
+}
+
+/*The function returns true, if the board is in a "winning state", otherwise return false
+winnings state means: you are restacked on either stacks b or c, biggest to smallest*/
+const checkForWin = () => {
+  // Your code here
+  if(stacks.b.length === 4){
```

Comment:
the comments mentions a win is either all the elements are on b, or on c, but the code only checks b/